src/dag/read.ts

```diff
@@ -0,0 +1,35 @@
+import type * as kv from '../kv/mod.js';
+import {Chunk} from './chunk.js';
+import {chunkDataKey, chunkMetaKey, headKey} from './key.js';
+import * as utf8 from '../utf8.js';
+
+export class Read {
+  private readonly _kvr: kv.Read;
+
+  constructor(kv: kv.Read) {
+    this._kvr = kv;
+  }
+
+  async hasChunk(hash: string): Promise<boolean> {
+    return await this._kvr.has(chunkDataKey(hash));
+  }
+
+  async getChunk(hash: string): Promise<Chunk | undefined> {
+    const data = await this._kvr.get(chunkDataKey(hash));
+    if (data === undefined) {
+      return undefined;
+    }
+
+    const meta = await this._kvr.get(chunkMetaKey(hash));
```

Comment:
Yeah I feel like this was overly clever now that I understand perf characteristics of IDB better. I bet we are really doing ourselves a disservice storing this tiny bit of data separately.